### PR TITLE
Add a version filter to enable bwc in 2.12

### LINKED ISSUE #1838: [Bug] BWC Rolling Upgrade Tests are failing on neural search due to changes in MLPredictionTaskRequest Object
**What is the bug?**
The BWC Rolling upgrade tests are failing in the neural search because of recent [changes](https://github.com/opensearch-project/ml-commons/commit/bab9439e17c98429f4bd9f1ac853c1db20ef3219#diff-c7e3f7d91a1e68b0bee8c7397ec92bf7238296ac7ad8e61195a94998e4929fef) in ML-Commons.  
So the scenario is when the bwc version is 2.10 and current version is 2.12.0-SNAPSHOT
```
./gradlew :qa:rolling-upgrade:testAgainstOneThirdUpgradedCluster -Dbwc.version=2.10.0 
```
The tests fails with a reason
```
org.opensearch.transport.RemoteTransportException: [neuralSearchBwcCluster-rolling-1][127.0.0.1:35231][cluster:admin/opensearch/ml/predict]
Caused by: java.lang.IllegalStateException: unexpected byte [0x48]
	at org.opensearch.core.common.io.stream.StreamInput.readBoolean(StreamInput.java:592) ~[opensearch-core-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.core.common.io.stream.StreamInput.readBoolean(StreamInput.java:582) ~[opensearch-core-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.ml.common.transport.prediction.MLPredictionTaskRequest.<init>(MLPredictionTaskRequest.java:56) ~[?:?]
	at org.opensearch.transport.RequestHandlerRegistry.newRequest(RequestHandlerRegistry.java:85) ~[opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.InboundHandler.newRequest(InboundHandler.java:283) [opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.InboundHandler.handleRequest(InboundHandler.java:243) [opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.InboundHandler.messageReceived(InboundHandler.java:133) [opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.InboundHandler.inboundMessage(InboundHandler.java:115) [opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.TcpTransport.inboundMessage(TcpTransport.java:767) [opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:175) [opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:150) [opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:115) [opensearch-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at org.opensearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:95) [transport-netty4-client-2.12.0-SNAPSHOT.jar:2.12.0-SNAPSHOT]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.101.Final.jar:4.1.101.Final]
```

It is because there is one more constructor for MLPredictionTaskRequest has been added in the PR mentioned above.

In BWC tests when cluster has 3 nodes and 2 of them are running on 2.10 and one node gets upgraded to 2.12.0-SNAPSHOT. Now 
when neural search calls predict api then while initializing the MLPredictionTaskRequest [here](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java#L68C22-L68C22) it fails at this [line](https://github.com/opensearch-project/ml-commons/blob/d18ab03b9d09e198c347389acfd59c86ca4346f5/common/src/main/java/org/opensearch/ml/common/transport/prediction/MLPredictionTaskRequest.java#L56C13-L56C13) with error mentioned above.

It is happening because it is either getting more parameters or no parameters for boolean while executing in.readBoolean().

 The same is happening when the BWC tests are executed for 2.9 OS version.

The tests are running fine for 2.11 OS version. 
But it makes this tests execution flaky.
Because if Out of 3 nodes if it uses the node which is running 2.11 version for calling predict api rather than 2.12.0-SNAPSHOT then it will face the same issue.

**How can one reproduce the bug?**
1. Clone the neural search project
2. checkout on 2.x branch
3.  Run ./gradlew :qa:rolling-upgrade:testAgainstOneThirdUpgradedCluster -Dbwc.version=2.10.0 

**What is the expected behavior?**
The BWC rolling upgrade tests should run with all previous version with no error.

**What is your host/environment?**
 - OS: linux
 - Version  
 - Plugins  Neural Search

### PR DESCRIPTION
**Pull request by b4sjoo**
### Description
This will fix the breaking change introduced by 2.11
 
### Issues Resolved
This will partly resolved https://github.com/opensearch-project/ml-commons/issues/1838
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java [hunk 1]
```diff
@@ -530,7 +530,7 @@ private void indexRemoteModel(
                     .build();
 
                 IndexRequest indexModelMetaRequest = new IndexRequest(ML_MODEL_INDEX);
-                if (registerModelInput.getIsHidden()) {
+                if (registerModelInput.getIsHidden() != null && registerModelInput.getIsHidden())  {
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java:533]: Check if any other places using `registerModelInput.getIsHidden()` and add null checking?

**Comment by ylwu-amzn** [comment]
```
* What went wrong:
Execution failed for task ':opensearch-ml-plugin:spotlessJavaCheck'.
> The following files had format violations:
      src/main/java/org/opensearch/ml/model/MLModelManager.java
          @@ -530,7 +530,7 @@
           ????????????????????.build();
           
           ????????????????IndexRequest?indexModelMetaRequest?=?new?IndexRequest(ML_MODEL_INDEX);
          -????????????????if?(registerModelInput.getIsHidden()?!=?null?&&?registerModelInput.getIsHidden())??{
          +????????????????if?(registerModelInput.getIsHidden()?!=?null?&&?registerModelInput.getIsHidden())?{
           ????????????????????indexModelMetaRequest.id(modelName);
```

File: common/src/main/java/org/opensearch/ml/common/dataset/TextDocsInputDataSet.java [hunk 2]
```diff
@@ -18,6 +19,8 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.regex.Matcher;
```
Comment (ylwu-amzn) [inline common/src/main/java/org/opensearch/ml/common/dataset/TextDocsInputDataSet.java:22]: Seems not being used, remove ?

Comment (b4sjoo) [reply common/src/main/java/org/opensearch/ml/common/dataset/TextDocsInputDataSet.java]: Removed

File: plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java (hunk 1 above)
Comment (b4sjoo) [reply plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java]: Found another one, added.

**Review by ylwu-amzn — APPROVED** [review]

**Review by jngz-es — APPROVED** [review]

**Review by mingshl — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

**Pull request merged**